ghcide/src/Development/IDE/Core/FileStore.hs

```diff
@@ -121,6 +124,19 @@ getModificationTimeRule vfs isWatched =
                             then return (Nothing, ([], Nothing))
                             else return (Nothing, ([diag], Nothing))
 
+-- | Interface files cannot be watched, since they live outside the workspace.
+--   But interface files are private, in that only HLS writes them.
+--   So we implement watching ourselves, and bypass the need for alwaysRerun.
+isInterface :: NormalizedFilePath -> Bool
+isInterface f = takeExtension (fromNormalizedFilePath f) `elem` [".hi", ".hie"]
+
+-- | Reset the GetModificationTime state of interface files
+resetInterfaceStore :: ShakeExtras -> FilePath -> IO ()
+resetInterfaceStore state f = do
+    forM_ [toNormalizedFilePath' (replaceExtension f ext) | ext <- ["hi","hie"]] $ \f ->
```

Comment:
Do we still need to replace extensions? I thought it was only called with the `.hi` file. Also, this will overwrite the extension for `hi-boot` files.

Reply:
Good point, thanks! Not needed and I'll update `isInterface` to account for `.hi-boot` files too